protocol/bc/spend.go

```diff
@@ -12,6 +19,29 @@ type SpendInput struct {
 
 func (si *SpendInput) IsIssuance() bool { return false }
 
+func (si *SpendInput) writePrevoutCommitment(w io.Writer, assetVersion uint64) error {
+	return si.OutputCommitment.writeTo(w, assetVersion)
+}
+
+// readCommitment reads a spend input commitment AFTER the leading
+// type byte has been consumed.
+func (si *SpendInput) readCommitment(r io.Reader, txVersion, assetVersion uint64) (err error) {
```

Comment:
Perhaps a comment at the call site would be enough--I'm basically looking for ways to reduce cognitive load while reading this code.